user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java

```diff
@@ -77,7 +77,7 @@ public void createOrUpdate(String username, KafkaUserQuotas quotas) {
             JsonNode diff = null;
             try {
                 ObjectMapper objectMapper = new ObjectMapper();
-                diff = JsonDiff.asJson(objectMapper.readTree(data), objectMapper.readTree(quotas.toString()));
+                diff = JsonDiff.asJson(objectMapper.readTree(data), objectMapper.readTree(createUserJson(quotas)));
```

Comment:
`createUserJson()` constructs some JsonNode and then serializes it to a String, which you're passing to `readTree`. I think you should instead factor a method out of `createUserJson()` which will return the `JsonNode`, and call that method directly here, avoiding the need to use `readTree()`.

Reply:
Looks like this is a mixture of vertx and jackson APIs, so it's fine for now.